C/plugins/storage/postgres/connection.cpp

```diff
@@ -1980,9 +1995,27 @@ bool Connection::jsonModifiers(const Value& payload, SQLBuffer& sql)
 				raiseError("Select sort", "Missing property \"column\"");
 				return false;
 			}
-			sql.append("\"");
-			sql.append(sortBy["column"].GetString());
-			sql.append("\"");
+
+			// Check wether column name in GROUP BY is the same
+			// of column name in ORDER BY
+			if (!groupColumn.empty() &&
+			    groupColumn.compare(sortBy["column"].GetString()) == 0 &&
+			    nAggregates)
+			{
+				// Increment number of columns
+				// as the GROUP BY column is added as last one
+				// in the column names for SELECT
+				// The ORDER BY column name is replaced by the column
+				// number, without diuble quotes
+				// example: SELECT ... ORDER BY 4
+				sql.append(++nAggregates);
```

Comment:
I would have used nAggregates + 1 here and not done the increment as it means nAggregates is now a misleading number. But it works intros case as we never use nAggregates again.

Reply:
Fixed